modules/client-configuration/pages/contact-methods-saltssh-tunnel.adoc

```diff
@@ -0,0 +1,97 @@
+
+The server uses SSH to contact the clients at regular intervals, checking in and performing scheduled actions and events.
+
+
+// FiXME: only here with tunnel, otherwise move to contact-methods-saltssh.adoc
+[IMPORTANT]
+====
+Re-installing systems using the provisioning model is not currently supported on clients managed with SSH Push.
+====
+
+
+[IMPORTANT]
+====
+The tunnel is used to provide the access to the server through the encrypted tunnel.
+The repositories assigned to the SSH Push client (with tunnel) are provided through this tunnel only, thus it is not possible to use package manager tools directly from the client system because the repositories are available only while the tunnel is up.
+In other words, access is only possible if the session is initiated by the server.
+All package managing operations on the client can be performed from the server side only.
+====
+
+For tunneling connections via SSH, a port number is required for tunneling via HTTPS.
+The port number used by default is `1233`.
+To overwrite it, you can add a custom port numbers greater than 1024 to [path]``/etc/rhn/rhn.conf``:
+
+----
+ssh_push_port_https = high_port
+----
+
+
+For security reasons, you might want to use sudo with SSH, to access the system as an unprivileged user instead of as root.
+
+
+.Procedure: Configuring Unprivileged SSH Access
+. Ensure you have the latest [path]``spacewalk-taskomatic`` and [path]``spacewalk-certs-tools`` packages installed on the {productname} Server.
+. On each client system, create an appropriate unprivileged user.
+. On each client system, edit the [filename]``sudoers`` file:
++
+----
+sudo visudo
+----
+. Grant [command]``sudo`` access to the user by adding this line at the end of the [filename]``sudoers`` file.
+  Replace [systemitem]``<user>`` with the name of the user that is bootstrapping the client in the {webui}:
++
+----
+<user>  ALL=NOPASSWD: /usr/bin/python, /usr/bin/python2, /usr/bin/python3, /var/tmp/venv-salt-minion/bin/python
+----
++
+[NOTE]
+====
+This procedure grants root access without requiring a password, which is required for registering the client.
+When the client is successfully installed it runs with root privileges, so the access is no longer required.
+We recommend that you remove the line from the [path]``sudoers`` file after the client has been successfully installed.
+====
+
+. On the {productname} Server, in the [path]``/etc/rhn/rhn.conf`` configuration file, add or amend this line to include the unprivileged username:
++
+----
+ssh_push_sudo_user = <user>
+----
+
+
+You need to use the {webui} or API to register these clients with the {productname} Server.
+
+
+For more information about bootstrapping, see xref:client-configuration:registration-bootstrap.adoc[].
+
+Before you begin, you need to ensure that you have specified which ports to use for SSH tunneling.
+If you have registered clients before changing the port numbers, they need to be registered again.
```

Comment:
This note could be easily missed while reading, but it's important point. Does anyone know if bootstrapping with reactivation key will work for such case? If so, maybe put more visible note about it here?

Reply:
I tried bootstraping a Salt SSH client with reactivation key and it worked (the venv-salt-minion service is running in the client). However when I get the system details (using spacecmd) it has still the value` "contact_method": "ssh-push"`.

Reply:
I also have to check of course if a port change is feasible when the client was already bootstrapped.

Reply:
That seems to work.

Reply:
ok, thanks for testing!